client/extensions/woocommerce/app/products/product-create.js

```diff
@@ -93,22 +94,29 @@ class ProductCreate extends React.Component {
 	onSave = () => {
 		const { site, product, translate } = this.props;
 
-		const successAction = () => {
+		const successAction = ( products ) => {
+			const newProduct = head( products );
 			page.redirect( getLink( '/store/products/:site', site ) );
-
 			return successNotice(
 				translate( '%(product)s successfully created.', {
 					args: { product: product.name },
 				} ),
-				{ duration: 4000, isPersistent: true }
+				{
+					displayOnNextPage: true,
+					duration: 8000,
+					button: translate( 'View' ),
+					onClick: () => {
+						window.open( newProduct.permalink );
+					},
+				}
 			);
 		};
 
 		const failureAction = errorNotice(
 			translate( 'There was a problem saving %(product)s. Please try again.', {
 				args: { product: product.name },
 			} ),
-			{ isPersistent: true },
+			{ displayOnNextPage: true },
```

Comment:
I also logged an issue for better error messages... we can definitely detect a few different things and be more helpful here.